lighthouse-core/audits/dobetterweb/has-inspector-issues.js

```diff
@@ -0,0 +1,238 @@
+    'a Cross-Origin Opener Policy was blocked',
+  /** Message shown in a data table when the item is a HeavyAds issue where an ad uses more than 4 megabytes of network bandwith. */
+  heavyAdsNetworkLimitMessage: 'The page contains ads that use ' +
+    'more than 4 megabytes of network bandwidth',
+  /** Message shown in a data table when the item is a HeavyAds issue where an ad has used the main thread for more than 60 seconds in total. */
+  heavyAdsCPUTotalLimitMessage: 'The page contains ads that use the ' +
+    'main thread for more than 60 seconds in total',
+  /** Message shown in a data table when the item is a HeavyAds issue where an ad has used the main thread for more than 15 seconds in any 30 second window. */
+  heavyAdsCPUPeakLimitMessage: 'The page contains ads that use the ' +
+    'main thread for more than 15 seconds in a 30 second window',
+  /** Message shown in a data table when the item is a ContentSecurityPolicy issue where resources are blocked due to not being in the Content Security Policy header. */
+  cspUrlViolationMessage: 'The CSP of the page blocks some resources because ' +
+    'their origin is not included in the content security policy header',
+  /** Message shown in a data table when the item is a ContentSecurityPolicy issue where the Content Security Policy blocks inline execution of scripts and stylesheets. */
+  cspInlineViolationMessage: 'The CSP of the page blocks ' +
+    'inline execution of scripts and stylesheets',
+  /** Message shown in a data table when the item is a ContentSecurityPolicy issue where the Content Security Policy blocks the use of the `eval` function in Javascript. */
+  cspEvalViolationMessage: 'The CSP of the site blocks ' +
+    'the use of `eval` in JavaScript',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/** @type {Object<string, string>} */
+const heavyAdsMsgMap = {
+  'NetworkTotalLimit': str_(UIStrings.heavyAdsNetworkLimitMessage),
+  'CpuTotalLimit': str_(UIStrings.heavyAdsCPUTotalLimitMessage),
+  'CpuPeakLimit': str_(UIStrings.heavyAdsCPUPeakLimitMessage),
+};
+/** @type {Object<string, string>} */
+const contentSecurityPolicyMsgMap = {
+  'kInlineViolation': str_(UIStrings.cspInlineViolationMessage),
+  'kEvalViolation': str_(UIStrings.cspEvalViolationMessage),
+  'kURLViolation': str_(UIStrings.cspUrlViolationMessage),
+};
+/** @type {Object<string, string>} */
+const blockedByResponseMsgMap = {
+  'CoepFrameResourceNeedsCoepHeader': str_(UIStrings.coepResourceBlockedMessage),
+  'CoopSandboxedIFrameCannotNavigateToCoopPage': str_(UIStrings.coopIframeBlockedMessage),
+  'CorpNotSameOrigin': str_(UIStrings.coepResourceBlockedMessage),
+  'CorpNotSameOriginAfterDefaultedToSameOriginByCoep': str_(UIStrings.coepFrameBlockedMessage),
+  'CorpNotSameSite': str_(UIStrings.coepResourceBlockedMessage),
+};
+
+/** @typedef {{issueType: string, description: string, requestUrl?: string}} IssueItem */
```

Comment:
move typedefs to top of file (before requires) for consistency with other files + easy discovery.